diff --git a/node_modules/@react-navigation/native/lib/module/useLinking.js b/node_modules/@react-navigation/native/lib/module/useLinking.js
index 3321728..62c5fef 100644
--- a/node_modules/@react-navigation/native/lib/module/useLinking.js
+++ b/node_modules/@react-navigation/native/lib/module/useLinking.js
@@ -470,6 +470,13 @@ export default function useLinking(ref, {
 
       const previousState = previousStateRef.current;
       const state = navigation.getRootState();
+
+      // This can happen if root navigator is swapped, for example with user login?
+      // https://github.com/react-navigation/react-navigation/issues/10185
+      if (!state) {
+        return;
+      }
+
       const pendingPath = pendingPopStatePathRef.current;
       const route = findFocusedRoute(state);
       const path = (_route$path2 = route === null || route === void 0 ? void 0 : route.path) !== null && _route$path2 !== void 0 ? _route$path2 : getPathFromStateRef.current(state, configRef.current);
diff --git a/node_modules/@react-navigation/native/src/useLinking.tsx b/node_modules/@react-navigation/native/src/useLinking.tsx
index bda378e..67d00de 100644
--- a/node_modules/@react-navigation/native/src/useLinking.tsx
+++ b/node_modules/@react-navigation/native/src/useLinking.tsx
@@ -510,6 +510,12 @@ export default function useLinking(
       const previousState = previousStateRef.current;
       const state = navigation.getRootState();
 
+      // This can happen if root navigator is swapped, for example with user login?
+      // https://github.com/react-navigation/react-navigation/issues/10185
+      if (!state) {
+        return;
+      }
+
       const pendingPath = pendingPopStatePathRef.current;
       const route = findFocusedRoute(state);
       const path =
